activesupport/lib/active_support/cache/strategy/local_cache.rb

```diff
@@ -108,6 +112,12 @@ def decrement(name, amount = 1, **options) # :nodoc:
           value
         end
 
+        def exist?(name, options = nil) # :nodoc:
```

Comment:
Should `read_serialized_entry` be modified instead to not write a value when we are trying to read (either through `exist?` or `read`)?

This behaviour is otherwise similar and questionable:

```ruby
cache.with_local_cache do
  cache.read("foo") # nil
  cache.fetch_multi("foo") { "bar" } # { "foo" => nil }
end
```

Reply:
If I understood yours suggestion correctly, we need to update existing `read` and `exist?` methods to pass some internal value indicating for local cache that do not store the result? That approach looks complex for me, given that the local cache does not even directly implement these methods.

I implemented a simpler solution - don't locally cache `nil`s. This behaviour change doesn't look problematic for me.

Reply:
Yeah you are right, nevermind me, after having digger more look this seems complex for no reasons. But it looks like caching nil value was on purpose (https://github.com/rails/rails/pull/22194) for faster cache miss, I don't think we'd want to change that.

Perhaps `exist?` should be treated differently and your original commit was enough.
The behaviour in my snippet above seems off but I don't really have a solution.